### ISSUE TITLE
Question: Any plans to port this to VSC?

### ISSUE THREAD

**Issue by compostbrain** [t9md/atom-vim-mode-plus#1063]
I would like to have this available in Visual Studio Code.  There is a Vim emulation extension but it seems to be missing some features I enjoy in this package.

**Comment by t9md** [comment]
Currently no. But I think if you think vmp is better tha VsCodeVim, putting that impression to them might lead them to do more research on vmp. Which is very honor to me as one vim-plugin author among tons of similar projects.

**Comment by t9md** [comment]
Thanks, and what you've suggested is just jordan suggested long before.

https://github.com/t9md/atom-vim-mode-plus/issues/182

Theoretically it's possible, and what vmp shines is each operator/motion is very declaratively defined which might make editor-agnostic abstraction easy.
But still vmp greatly depends on Atom's super granular APIs.
And I myself am happy with Atom. Feeling nor particular motivation to move to VsCode.
If I forced to use VSCode, then I might consider to implement it on top of VSCode or collaborate with already existing VSCodeVim.

**Comment by calebmeyer** [comment]
The problem with VS Code (and unfortunately with VS Code Vim) is that there's not enough information in the public API to enable a fully compatible (or even as good as atom's) vim extension. See https://github.com/Microsoft/vscode/issues/22276#issuecomment-302061827 for more details on this. Because of the architecture of VS Code, extensions cannot know some of the things the core does.

I agree it would be cool if vim-mode-plus were editor agnostic, but I don't think VS Code will (soon, or maybe ever) support all the API needed to do it.

**Comment by cquiroz** [comment]
This is a very good discussion. For me, it would be better to use vscode as scala integration is better. But, vmp is so good that it keeps me firmly in Atom

Thanks @t9md for the best vim emulation I've used

**Comment by bennypowers** [comment]
VMP is a beast. I'm totally hooked. Thank you @t9md for producing this excellent software.

With Facebook shutting down nuclide and Atom-IDE, and with VSCode growing in popularity, I'm worried that a day might come in the future when atom is no longer relevant, and I'll have to give up on `c o i p`, etc., which have become integral to my workflow.

I hope work on abstracting the core of VMP will succeed so that many more users can enjoy these features.

**Comment by t9md** [comment]
Oh, I didn’t know FB shut down nuclide.
Surprised, and sad.
For me, I’m really looking forward to see xray editor become generally available.
Hopefully vmp works on top of xray with slight modifications.
In case that will not happen and Atom is also decided to shut down, the I think I can create vmp for VSCode then!!

**Comment by calebmeyer** [comment]
Yeah, I'd switch to xray in a heartbeat if it's as fast as what's out there make it seem. It would be the best of atom and sublime text.

I don't think atom will be shut down any time soon, per the community manager, Lee: https://discuss.atom.io/t/the-microsoft-intervention/55744

If it does get shut down, I hope VS Code will be able to support all the cool features of VMP that aren't in vscodevim
- additional text objects like inner entire (`y i e` is 100% my most used shortcut, so much nicer than `ggVGy` and leaves your cursor in place)
- Flash on changes made is so nice
- Occurrences are awesome too

**Comment by deepsweet** [comment]
Lack of subword movements in VSCode is so critical for me, as well as occurrences. But I personally have no choice because I work a lot with TypeScript recently and VSCode works just flawlessly with it.

Would love to see VMP ported to VSCode ❤️

**Comment by mauricecruz** [comment]
I'm still using Atom ... but I fear that they're going to shut it down in the near future. Development has slowed significantly in the last half of the year.

![image](https://user-images.githubusercontent.com/1640686/67948769-cb9d0700-fbbc-11e9-95ce-d98e0478221f.png)

**Comment by t9md** [comment]
Is that Atom's activity right?

**Comment by Spirarel** [comment]
https://github.com/atom/atom/graphs/commit-activity

**Comment by mauricecruz** [comment]
Yes @t9md. Thanks for posting the link @Spirarel.

**Comment by kerwin6182828** [comment]
> xray

so love you

**Comment by bennypowers** [comment]
Note for anyone coming here from Google, if you have atom-ide-ui and atom-ide-typescript, but not atom-typescript installed, you can bind `g d` to hyperclick word under cursor with:

```cson
'atom-text-editor.vim-mode-plus.normal-mode':
  'g d':      'hyperclick:confirm-cursor'
  'g h':      'datatip:toggle'
```

**Comment by bennypowers** [comment]
this might be of interest to the discussion

https://theia-ide.org/

**Comment by bennypowers** [comment]
Also of interest is https://github.com/asvetliakov/vscode-neovim which bypasses `type` bindings

**Comment by keevan** [comment]
For what it's worth, github announced the sunsetting of Atom: https://github.blog/2022-06-08-sunsetting-atom/

**Comment by t9md** [comment]
@keevan Thank you for sharing this. It is sad. very sad.

**Comment by saadq** [comment]
> @keevan Thank you for sharing this. It is sad. very sad.

Indeed. Thanks so much for creating this plugin though, it was by far the best Vim integration I've used in any editor/IDE by far. Atom + vim-mode-plus will be missed.

The bright side is that some Atom devs are working on a new editor called Zed (https://zed.dev). Hoping that will be a good successor
